torch/csrc/api/include/torch/nn/functional/loss.h

```diff
@@ -10,37 +11,104 @@ inline Tensor l1_loss(
     const Tensor& input,
     const Tensor& target,
     const L1LossOptions& options = {}) {
-  return torch::l1_loss(input, target, options.reduction());
+  return torch::l1_loss(
+    input,
+    target,
+    c10::visit(enumtype::_reduction_get_enum{}, options.reduction()));
```

Comment:
Instead of manually typing out `c10::visit` everywhere, why not come up with a good API for doing this and call that instead? Especially since `_reduction_get_enum` is underscored...

Reply:
Thanks for the suggestion! I added `torch::enumtype::reduction_get_enum()` API for this purpose :D